assets/pages/monthly-contributions/reducers/reducers.js

```diff
@@ -18,54 +25,59 @@ import type { Action } from '../actions/monthlyContributionsActions';
 
 export type State = {
   amount: number,
-  country: string,
+  currency: Currency,
+  country: IsoCountry,
   error: ?string,
   payPalType: PayPalButtonType,
 };
 
-
-// ----- Setup ----- //
-
-const initialState: State = {
-  amount: 5,
-  country: 'GB',
-  error: null,
-  payPalType: 'NotSet',
+export type CombinedState = {
+  monthlyContrib: State,
+  intCmp: string,
+  user: UserState,
+  stripeCheckout: StripeCheckoutState,
+  payPalExpressCheckout: PayPalExpressCheckoutState,
+  csrf: CsrfState,
 };
 
-
 // ----- Reducers ----- //
 
-function monthlyContrib(
-  state: State = initialState,
-  action: Action): State {
-
-  switch (action.type) {
+function monthlyContrib(amount: number, currency: Currency, country: IsoCountry) {
 
-    case 'SET_CONTRIB_VALUE':
-      return Object.assign({}, state, { amount: action.value });
+  const initialState: State = {
+    amount,
+    currency,
+    country,
+    error: null,
+    payPalButtonExists: false,
+    payPalType: 'NotSet',
+  };
 
-    case 'CHECKOUT_ERROR':
-      return Object.assign({}, state, { error: action.message });
+  return (state: State = initialState, action: Action): State => {
+    switch (action.type) {
 
-    case 'SET_PAYPAL_BUTTON' :
-      return Object.assign({}, state, { payPalType: action.value });
+      case 'CHECKOUT_ERROR':
+        return Object.assign({}, state, { error: action.message });
 
-    default:
-      return state;
+      case 'SET_PAYPAL_BUTTON' :
+        return Object.assign({}, state, { payPalType: action.value });
 
-  }
+      default:
+        return state;
 
+    }
+  };
 }
 
 
 // ----- Exports ----- //
 
-export default combineReducers({
-  monthlyContrib,
-  intCmp,
-  user,
-  stripeCheckout,
-  payPalExpressCheckout,
-  csrf,
-});
+export default (amount: number, currency: Currency, country: IsoCountry) =>
+  combineReducers({
```

Comment:
And the result of `combineReducers` would be the ***RootMonthlyContributionsReducer***